### ISSUE TITLE
Suggestion: complain about  var=$"(  typos 

### ISSUE THREAD

**Issue by eastlondoner** [koalaman/shellcheck#1516]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash

foo(){
        echo "bar"
}
accidental_typo=$"(foo)"
what_i_meant="$(foo)"

[[ "${accidental_typo}" == "${what_i_meant}" ]] && echo "darn"

```

#### Here's what shellcheck currently says:

```No issues detected!```

#### Here's what I wanted or expected to see:

`var=$"(` doesn't make any sense as a variable assignment. Perhaps this is a typo (did you mean `var="$(` ?

**Comment by koalaman** [comment]
Duplicate of #1517
